includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -122,33 +127,170 @@ public static function handle_password_reset_request() {
+
+		$switched_locale = \switch_to_locale( get_user_locale( $user ) );
+
+		$email = [
+			'to'      => $user->user_email,
+			/* translators: %s Site title. */
+			'subject' => __( '[%s] Account deletion request', 'newspack' ),
+			'message' => $message,
+			'headers' => [
+				sprintf(
+					'From: %1$s <%2$s>',
+					Reader_Activation::get_from_name(),
+					Reader_Activation::get_from_email()
+				),
+			],
+		];
+
+		/**
+		 * Filters the account deletion email.
+		 *
+		 * @param array    $email Email arguments. {
+		 *   Used to build wp_mail().
+		 *
+		 *   @type string $to      The intended recipient - New user email address.
+		 *   @type string $subject The subject of the email.
+		 *   @type string $message The body of the email.
+		 *   @type string $headers The headers of the email.
+		 * }
+		 * @param \WP_User $user  User to send the magic link to.
+		 * @param string   $url   Account deletion form url.
+		 */
+		$email = \apply_filters( 'newspack_reader_account_delete_email', $email, $user, $url );
+
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_mail_wp_mail
+		$sent = \wp_mail(
+			$email['to'],
+			\wp_specialchars_decode( sprintf( $email['subject'], $blogname ) ),
+			$email['message'],
+			$email['headers']
+		);
+
+		if ( $switched_locale ) {
+			\restore_previous_locale();
+		}
+
+		if ( $sent ) {
+			wc_add_notice( __( 'Please check your email inbox for instructions on how to delete your account.', 'newspack' ), 'success' );
+		} else {
+			wc_add_notice( __( 'Something went wrong.', 'newspack' ), 'error' );
+		}
+
+		\wp_safe_redirect( \remove_query_arg( self::DELETE_ACCOUNT_URL_PARAM ) );
+		exit;
+	}
+
+	/**
+	 * Handle delete account confirmation.
+	 */
+	public static function handle_delete_account() {
+
+		/** Make sure `wp_delete_user()` is available. */
+		require_once ABSPATH . 'wp-admin/includes/user.php';
+
+		if ( ! isset( $_POST[ self::DELETE_ACCOUNT_FORM ] ) ) {
+			return;
+		}
+
+		$form_nonce = \sanitize_text_field( $_POST[ self::DELETE_ACCOUNT_FORM ] );
+		if ( ! $form_nonce || ! \wp_verify_nonce( $form_nonce, self::DELETE_ACCOUNT_FORM ) ) {
+			\wp_die( \esc_html__( 'Invalid request.', 'newspack' ) );
 		}
+
+		if ( ! isset( $_POST['confirm_delete'] ) ) {
+			return;
+		}
+
+		if ( ! \is_user_logged_in() ) {
+			return;
+		}
+
+		$user_id = \get_current_user_id();
+		$user    = \wp_get_current_user();
+		if ( ! Reader_Activation::is_user_reader( $user ) ) {
+			return;
+		}
+
+		$token           = isset( $_POST['token'] ) ? \sanitize_text_field( $_POST['token'] ) : '';
+		$transient_token = \get_transient( 'np_reader_account_delete_' . $user_id );
+		if ( ! $token || ! $transient_token || $token !== $transient_token ) {
+			wp_die( \esc_html__( 'Invalid request.', 'newspack' ) );
+		}
+		\delete_transient( 'np_reader_account_delete_' . $user_id );
+
+		\wp_delete_user( $user_id );
+
+		\wc_add_notice( __( 'Your account has been deleted.', 'newspack' ), 'success' );
```

Comment:
Nit: this message is never visible, since as soon as you confirm deletion of your account, you're dumped back to the login screen. Ideally we should be able to show notices in the login screen as well.

Reply:
11f8b180544a86025aefccf2f56abcd9130fb8ff fixes the notice strategy